LoopKit/DeviceManager/CGMManager.swift

```diff
@@ -111,6 +111,11 @@ public protocol CGMManager: DeviceManager {
     ///
     /// - Parameter observer: The observing object
     func removeStatusObserver(_ observer: CGMManagerStatusObserver)
+
+    /// Requests that the manager completes its deletion process
+    ///
+    /// - Parameter completion: Action to take after the CGM manager is deleted
+    func delete(completion: @escaping () -> Void)
```

Comment:
I added a similar function to `Service` in an earlier LoopKit PR related to Onboarding, but @ps2 disagreed with its necessity. See https://github.com/tidepool-org/LoopKit/pull/318#discussion_r567016337 for further details. I think you may want to get an okay from @ps2 for this addition before you merge it.

Reply:
@darinkrauss  hm.... that may be a while. Do you remember why he disagreed? 

The dexcomCGMManager cleanup is just to close the communication, which really isn't needed. If you think this code will block, I'll remove the code and create a task to check-in with Pete when he is available again.

What do you think?

Reply:
His concern is in the comment thread, but was effectively - if the user wants to delete the Service/CGM/Pump, then they should do so in the Service/CGM/Pump UI (i.e. settings UI) and any cleanup can be done there before the `*wantsDeletion` callback on the delegate is invoked.

Reply:
Ah... well this is only for QA/Dev. In the end, the user will not be able to delete the CGMManager... as least not until another is supported.

Reply:
agreed to make a ticket and discussion with Pete when he is available.

Reply:
Yeah, generally deletion/cleanup should happen from the UI provided by the CGMManager. There may be user interaction required, for example, for some devices. I'd like to know if we now have a use case were it's necessary for Loop to tell a manager to delete itself.

Reply:
@ps2 Current use case is limited to QA and dev to still allow switching CGM managers (the debug menu presents the delete CGM manager option). For Tidepool Loop v1.0, the user will not have a why to delete the Dexcom CGM manager.